nmdc_server/ingest/common.py

```diff
@@ -47,3 +50,14 @@ def extract_extras(
         if key not in fields and key not in exclude:
             values["annotations"][key] = extract_value(value)
     return values
+
+
+def merge_or_delete(ingest_db: Session, query):
+    logger = logging.getLogger()
+    for row in query:
+        try:
+            ingest_db.merge(row)
+            ingest_db.commit()
+        except IntegrityError:
+            logger.error("Error: data object with download history was removed.")
```

Comment:
`merge_or_delete` seems like a pretty generic function, but this error message is quite specific to the case of download history. Maybe we just need to rename this function if it's really just designed for this specific case?